src/cargo/ops/cargo_generate_lockfile.rs

```diff
@@ -122,6 +122,26 @@ pub fn update_lockfile(ws: &Workspace<'_>, opts: &UpdateOptions<'_>) -> CargoRes
             }
         }
 
+        // Mirror `--workspace` and never avoid workspace members.
+        // Filtering them out here so the above processes them normally
+        // so their dependencies can be updated as requested
+        to_avoid = to_avoid
+            .into_iter()
+            .filter(|id| {
+                for package in ws.members() {
+                    let member_id = package.package_id();
+                    // Skip checking the `version` because `previous_resolve` might have a stale
+                    // value.
+                    // When dealing with workspace members, the other fields should be a
```

Comment:
This fix looks fairly easy and straightforward, though I have a scenario in mind. If a member depends on its own old version on crates.io, but now seems it cannot update to the latest version?

Reply:
We should be protected against this with the source check. In fact, the source check should be sufficient, without a name check, because it points to the directory for the package.

Reply:
Fair. I missed the source check part